core/drivers/crypto/caam/hal/imx_6_7/hal_clk_mx7.c

```diff
@@ -11,7 +11,7 @@
 
 void caam_hal_clk_enable(bool enable)
 {
-	vaddr_t ccm_base = (vaddr_t)phys_to_virt(CCM_BASE, MEM_AREA_IO_SEC);
+	vaddr_t ccm_base = (vaddr_t)phys_to_virt(CCM_BASE, MEM_AREA_IO_SEC, 1);
```

Comment:
From [imx7-crm.h](https://github.com/OP-TEE/optee_os/blob/3.14.0/core/arch/arm/plat-imx/registers/imx7-crm.h#L32-L36), `CCM_CCGRx_{SET|CLR}()` macros do cross 4kB boundaries.   @cneveux, please cross-check this is fine.

Reply:
@etienne-lms 
You're right but this is not a problem in this case as it's a register set mapped on a section not on a small page.

Reply:
it's not really a matter of being mapped by section or small page. See the initial comment for this P-R, and more precisely:Issue explained in more details at #4616.

Reply:
In plat-imx, we use the following code
`register_phys_mem_pgdir(MEM_AREA_IO_SEC, [base address], ROUNDUP([size], CORE_MMU_PGDIR_SIZE));` 
to mapped each HW registers block without using the small page mapping. This create scatter `phys_mem_map` table.
Then the `phys_mem_map` table is mapped at boot time with the `core_init_mmu_map()` and I didn't see call to the `core_mmu_add_mapping()` which rebuild the area size function of the TLB information (which define the MMU page granularity).
Hence, except if I missed something the `register_phys_mem_pgdir()` with a length roundup to a section will result to only level 1 MMU Mapping which are section or supersection. 

Am I correct?

Reply:
I think you are correct. If you're confident with the implementation here, then i'm fine.